### DISCUSSION TITLE
Is there any offline docs?

### DISCUSSION THREAD

**Discussion by User-425** [ujjwalguptaofficial/JsStore#294] [Q&A]
I need an offline documentation, because sometimes, I work offline

**Comment by ujjwalguptaofficial** [comment]
Hey currently - no offline documentation is available. I am thinking about converting it into offline app but will take some time.

For now - i will suggest you to take a pdf print of the doc.

**Reply by surruk51** [reply]
I sat with the docs and the test files to try to build a cheat sheet. I have not tested everything and it is definitely open to improvement.
Attached as an .odt file
[jsstore cheat sheet.odt](https://github.com/ujjwalguptaofficial/JsStore/files/11584206/jsstore.cheat.sheet.odt)

**Reply by surruk51** [reply]
here it i in markdown
## connect

var conn=new JsStore.Connection(new Worker("scripts/jsstore.worker.js"))

## create/attach/alter database

conn.initDb({

"name": "*db name*",

"version": **1**|*integer*,

"tables": \[

{

"name": "*tablename*",

"columns": {

"*colname*": {

"primaryKey": ***false***,

"autoIncrement": ***false***,

"notNull": ***true***,

"dataType": "***string***"

}

},

"alter": {

"2": {

"modify": {

"*colname*": {

"primaryKey": ***false***,

"autoIncrement": ***false***,

"notNull": ***true***,

"dataType": "***string***"

} } } } } \]

})

## drop database

Retrieve one or more rows from table

> conn.dropDb()

## select clause

Retrieve one or more rows from table

"from": "*tablename*",

"where": "see *where clause*",

"distinct": **false**,

"ignoreCase": **false**,

"order": \[

{"by":"*colname*","type":"***asc***"},

{"by":"*colname*","type": "***asc***"}

\],

"skip": *starting row num*,

"limit": *num rows to select*,

"groupBy": \["*colname*","*colname*"\]

## where clause:

{

"*fieldname*": "*value*",

"*fieldname*": in:\["*value*","*value*"\]

"or": {

"*fieldname*":{"*operator*": "*value*"},

"*fieldname*":{"*operator*": "*value*"}

},

join: {

with: "*tablename*",

type: "*inner*",

on: "*fieldname*=*fieldname*",

as: {

"*resultname*":"*fieldname*",

"*resultname*":"*fieldname*"

} } }

operators: "\>","\<","\>=","\<=","-","!=","Like"

## aggregate clause

aggregate: {

"min": \["*fieldname*","*fieldname*"\],  
"max": \["*fieldname*","*fieldname*"\],  
"count": \["*fieldname*","*fieldname*"\],  
"sum": \["*fieldname*","*fieldname*"\],  
"average": \["*fieldname*","*fieldname*"\],  
"list": \["*fieldname*","*fieldname*"\],  
}

## create row(s)

conn.insert({

"into": "tablename",

"values": \[

"*row object*",

"*row object*"

\] } )

## update row(s)

conn.update ({

"in": "*tablename*",

"set": {

"*fieldname*": "*value*"

},

"where": "*see* *where clause*"

})

## delete row(s)

conn.remove( {

"from": "*tablename*",

"where": "*see* *where clause*"

})

## count row(s)

conn.count( {

"from": "*tablename*",

"where": "*see* *where clause*"

})

## get row(s)

conn.select("*see select clause"*)

## get union

returns all rows without duplicating

conn.union(\[

"*see select clause*",

"*see select clause*"

\])

## get intersection

returns only rows in both sets

conn.intersect(\[

"*see select clause*",

"*see select clause*"

\])

## remove all rows

deletes all rows from table.

conn.clear(“*tablename*”)

**Comment by surruk51** [comment]
I think there are issues with the WHERE clause in the above. Asking for help on that in a separate post.
